### PR TITLE
Third milestone feature

### PR DESCRIPTION
**Pull request by haraphat01**

### CODE REVIEW

**Comment by haraphat01** [comment]
This is our final project on the third milestone project for Tic Tac Toe. The game logic has been made and it's running as it supposed to.

- To run the game, you need to run `bin/main.rb`, after that, enter your desired name as the first player and select either X or O as the symbol to play with.

- Once the first step is done, you need to type in the second player's name, automatically, a symbol will be assigned to the second player.

- You can begin to play the game.

- Additionally, position 1 can now be selected. 

This game is made with only Ruby.

Kindly check for your kind review.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

Good job on this @haraphat01  and @DickensJuma  👍

However, let's fix this:
- [ ] Kindly provide a descriptive pull request message, edit this pull request and add a description above it.

- [ ] On the game it's not possible to place a mark at position 1 on the board, please fix this.

Kindly fix these issues and request another review👍
Feel free to contact me on slack(mike) if you need further clarification.

HAPPY CODING👨‍💻

File: bin/main.rb [hunk 1]
```diff
@@ -1,109 +1,71 @@
...
-end
-# used selection to save whatever the value the first player chooses
-def board
-  puts
-  puts '1|2|3'
-  puts  '-----'
-  puts  '4|5|6'
-  puts  '-----'
-  puts  '7|8|9'
+# rubocop:disable Layout/LineLength, Lint/RedundantCopDisableDirective,Layout/LineLength
```
Comment (santiago-dev96) [inline bin/main.rb:4]: Why are you disabling the line length cop? It is a good practice to keen your lines below 120 characters for readability.
```suggestion
```

File: bin/main.rb [hunk 2]
```diff
@@ -1,109 +1,71 @@
...
+
+require_relative('../lib/board.rb')
+require_relative('../lib/player.rb')
+require_relative('../lib/game.rb')
+
+puts '--------Welcome to Tic-Tac-Toe game--------'
+puts 'Please choose a name and favourite Symbol to use.'
+
+puts 'First player, please enter your name '
+player1 = gets.chomp.capitalize!
```
Comment (santiago-dev96) [inline bin/main.rb]: You don't need to use the bang at the end.
```suggestion
player1 = gets.chomp.capitalize
```

File: bin/main.rb [hunk 3]
```diff
@@ -1,109 +1,71 @@
...
+letter_one = ''
+until %w[X O].include?(letter_one)
+  puts "#{player1}, please choose between X and O to play with."
+  letter_one = gets.chomp.upcase!
```
Comment (santiago-dev96) [inline bin/main.rb]:
```suggestion
  letter_one = gets.chomp.upcase
```

File: bin/main.rb [hunk 4]
```diff
@@ -1,109 +1,71 @@
...
-input = gets.chomp.to_i
-# we forced the two players to make sure they enter a valid value of 0-9
-until input.between?(0, 9)
-  puts ' you did not choose correct move, choose again'
-  input = gets.chomp.to_i
+letter_two = letter_one == 'O' ? 'X' : 'O'
+player_one = Player.new(player1, letter_one)
 
+puts 'Second player\'s name'
+player2 = gets.chomp.capitalize!
```
Comment (santiago-dev96) [inline bin/main.rb]:
```suggestion
player2 = gets.chomp.capitalize
```

File: bin/main.rb [hunk 5]
```diff
@@ -1,109 +1,71 @@
...
+while [player1].include?(player2)
```
Comment (santiago-dev96) [inline bin/main.rb]:
```suggestion
while player1 == player2
```

File: bin/main.rb [hunk 6]
```diff
@@ -1,109 +1,71 @@
...
+  puts '#The name {player1} already exists.'
+  puts 'Second player\'s name'
+  player2 = gets.chomp.capitalize!
```
Comment (santiago-dev96) [inline bin/main.rb]:
```suggestion
  player2 = gets.chomp.capitalize
```

**Review by santiago-dev96 — CHANGES_REQUESTED** [review]
# Changes required :wrench:

You are doing a great job so far! I almost approve this PR, but there is a little thing that needs to be fixed.

1. When you confirm the prompt for starting a new game the numbers in the board ar displaced, look at the picture below.

Screenshot from 2020-07-08 08-20-59

I'm sure that if you fix that it is very likely that your PR will be approved next time. Happy coding!

You can find me in slack as **srodrig**.

**Comment by haraphat01** [comment]
Thanks for your review.

We've worked on the requested changes. The board is no more displacing the board numbers while on the verge to restart the game and also the bang method has been removed.

**Review by wathigo — APPROVED** [review]
### Approved :heavy_check_mark: 
You have done an amazing job on this project!
I enjoyed playing the game.

**You may proceed to the next project.**

**Good luck :four_leaf_clover: 
Simon Wathigo**

**Pull request merged**